### DISCUSSION TITLE
Duda de implementación parte 1

### DISCUSSION THREAD

**Discussion by IgnacioHqz** [IIC2133-PUC/2024-2#154] [Tarea 2] [answered]
Hola! Estoy teniendo una confusión sobre cómo abordar la implementación para la consulta IN-RECTANGLE usando kd-tree como aparece en la cápsula. 
Estoy algo complicado con cómo hacer que los puntos queden como hojas una vez que ya no se pueda hacer más particiones. En la cápsula de menciona que si un punto queda justo en una partición vertical, mandarlo a la derecha,  y si queda en una partición horizontal, mandarlo hacia abajo. Entiendo que antes de particionar hay que saber si el área tiene puntos, pero en caso de que si tenga, habría que buscar dentro del área que forman las particiones considerando a x desde la linea azul ([x1, es decir, incluyendolo) hasta el siguiente limite x2), excluyéndolo) e Y desde el limite inferior (sin incluirlo) hasta la linea verde incluyéndola ( y2] ) para así considerar los puntos que se movieron a la derecha y abajo? En ese caso, habría que buscar por rango una coordenada y luego filtrar el resultado por la otra coordenada? No sé si estaré bien.
Cualquier comentario se agradece! Muchas gracias
![duda](https://github.com/user-attachments/assets/d6a7312d-8736-4eb6-ac0e-c54ed7058325)

**Comment by CarlangaUC** [comment, marked as answer]
Hola!

La idea en general es que al estar haciendo estas particiones y empleando una estructura árbol, puedas tener todos los hijos en estas áreas que se te forman y por ende todo nodo hijo de una partición te asegura que esta en ese rango, detalles de implementación como por ejemplo si un punto cae en la línea vertical ira a la derecha o abajo dependerá de tu implementación, mientras seas consistente debería estar bien.
